### PR TITLE
Added note to prefix documentation for corner cases where cunumeric results can diverge from numpy

### PR DESCRIPTION
**Pull request by rkarim2**

### CODE REVIEW

File: cunumeric/module.py [hunk 1]
```diff
@@ -4671,6 +4671,19 @@ def cumprod(
...
+    Due to the parallel nature of cuNumeric implementation, with floating point
+    and complex types, the accuracy of the results can differ compared to
+    NumPy. Furthermore when boundary values such as inf occur they might not
+    propagate as expected. Consider the hypothetical float32 array
+    ``[3e+37, 1, 100, 0.01]`` doing a cumprod on it should result in
+    ``[3e+37, 3e+37, inf, inf]`` in Numpy, however in cunumeric, if this array
+    is internally partitioned such that partition 0 has ``[3e+37, 1]``  and
+    partition 1 has ``[100, 0.01]``, the end result of the cumprod turns out
+    being ``[3e+37, 3e+37, inf, 3e+37]``, a result that albeit more acurate, is
+    often not expected.
```
Comment (bryevdv) [inline cunumeric/module.py:4685]: I think this message could be tightened up with plainer/simpler phrasing. Maybe something like:

>     CuNumeric's parallel implementation may yield different results from NumPy
>     with floating point and complex types. For example, when boundary values
>     such as inf occur they may not propagate as expected. Consider the float32
>     array ``[3e+37, 1, 100, 0.01]``. NumPy's cumprod will return a result of
>     ``[3e+37, 3e+37, inf, inf]``. However, cuNumeric might internally partition
>     the array such that partition 0 has ``[3e+37, 1]``  and partition 1 has 
>     ``[100, 0.01]``, returning the result ``[3e+37, 3e+37, inf, 3e+37]``.

File: cunumeric/module.py [hunk 2]
```diff
@@ -4726,6 +4739,14 @@ def cumsum(
     --------
     numpy.cumsum
 
+    Notes
+    -----
+    Due to the parallel nature of cuNumeric implementation, with floating point
+    and complex types, the accuracy of the results can differ compared to
+    NumPy. Furthermore when boundary values such as inf occur they might not
+    propagate as expected. For a more detailed explanation check
+    cunumeric.cumprod
```
Comment (bryevdv) [inline cunumeric/module.py:4748]: Suggest:

>     CuNumeric's parallel implementation may yield different results from NumPy
>     with floating point and complex types. For example, when boundary values
>     such as inf occur they may not propagate as expected. For more explanation
>     see cunumeric.cumprod.

File: cunumeric/module.py [hunk 3]
```diff
@@ -4646,12 +4646,12 @@ def cumprod(
...
     axis : int, optional
-        Axis along which the cumulative sum is computed. The default (None) is
-        to compute the cumsum over the flattened array.
+        Axis along which the cumulative product is computed. The default (None)
+        is to compute the cumprod over the flattened array.
 
     dtype : dtype, optional
         Type of the returned array and of the accumulator in which the elements
-        are summed. If dtype is not specified, it defaults to the dtype of a,
+        are returned. If dtype is not specified, it defaults to the dtype of a,
```
Comment (magnatelee) [inline cunumeric/module.py:4654]: I think this should be "multiplied"

File: cunumeric/module.py [hunk 4]
```diff
@@ -4706,7 +4719,7 @@ def cumsum(
...
```
Comment (magnatelee) [inline cunumeric/module.py:4722]: this edit doesn't seem to make much sense.

File: cunumeric/module.py (hunk 1 above)
Comment (rkarim2) [reply cunumeric/module.py]: a3a8593fed3c1f0534e2d2dc251d75967b9790c3

File: cunumeric/module.py (hunk 3 above)
Comment (rkarim2) [reply cunumeric/module.py]: a3a8593fed3c1f0534e2d2dc251d75967b9790c3

File: cunumeric/module.py (hunk 4 above)
Comment (rkarim2) [reply cunumeric/module.py]: a3a8593fed3c1f0534e2d2dc251d75967b9790c3

File: cunumeric/module.py (hunk 2 above)
Comment (rkarim2) [reply cunumeric/module.py]: a3a8593fed3c1f0534e2d2dc251d75967b9790c3

**Review by magnatelee — APPROVED** [review]

**Pull request merged**